lib/elixir/lib/module/checker.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Module.Checker do
+  def verify(module_map, _binary) do
+    state = :maps.with([:module, :file, :compile_opts], module_map)
+
+    module_map.definitions
+    |> Enum.reverse()
+    |> check_definitions(state)
+    |> List.flatten()
+  end
+
+  defp check_definitions(definitions, state) do
+    Enum.map(definitions, &check_definition(&1, state))
+  end
+
+  defp check_definition({def, _kind, meta, clauses}, state) do
+    state = :maps.put(:function, def, put_file_meta(state, meta))
+    Enum.map(clauses, &check_clause(&1, state))
+  end
+
+  defp put_file_meta(state, meta) do
+    case Keyword.fetch(meta, :file) do
+      {:ok, {file, line}} ->
+        :maps.merge(state, %{file: file, line: line})
```

Comment:
According to the above we should set only the file here: `%{state | file: file}`.